app/src/main/java/org/wikipedia/util/ThrowableUtil.kt

```diff
@@ -91,26 +94,30 @@ object ThrowableUtil {
     }
 
     @JvmStatic
-    fun parseBlockedError(info: MwServiceError.BlockInfo): String {
-        val app = WikipediaApp.getInstance()
-        var retStr = "<b>${app.getString(R.string.error_blocked_title)}</b>"
-        val title = PageTitle(UserAliasData.valueFor(app.appOrSystemLanguageCode), info.blockedBy, app.wikiSite)
+    @WorkerThread
+    fun getBlockMessageHtml(blockInfo: MwServiceError.BlockInfo): String {
+        var html = ""
+        Observable.zip(ServiceFactory.get(WikipediaApp.getInstance().wikiSite).userInfo,
+            ServiceFactory.get(WikipediaApp.getInstance().wikiSite).parsePage("MediaWiki:Blockedtext"),
+            ServiceFactory.get(WikipediaApp.getInstance().wikiSite).parseText(blockInfo.blockReason),
+            { userInfoResponse: MwQueryResponse, blockedParseResponse: MwParseResponse, reasonParseResponse: MwParseResponse ->
```

Comment:
Explicit type specifications can be removed.

Reply:
done!